src/menews_jisan.c

```diff
@@ -0,0 +1,199 @@
+                "\tldrb r2, [r5]\n"
+                "\tmovs r0, 0x4\n"
+                "\tnegs r0, r0\n"
+                "\tands r0, r2\n"
+                "\torrs r0, r1\n"
+                "\tstrb r0, [r5]\n"
+                "\tcmp r4, 0x2\n"
+                "\tbhi _08146C56\n"
+                "\tcmp r4, 0x1\n"
+                "\tbcs _08146C5C\n"
+                "\tb _08146C80\n"
+                "_08146C56:\n"
+                "\tcmp r4, 0x3\n"
+                "\tbeq _08146C6E\n"
+                "\tb _08146C80\n"
+                "_08146C5C:\n"
+                "\tbl Random\n"
+                "\tlsls r0, 16\n"
+                "\tlsrs r0, 16\n"
+                "\tmovs r1, 0xF\n"
+                "\tbl __umodsi3\n"
+                "\tadds r0, 0x10\n"
+                "\tb _08146C7E\n"
+                "_08146C6E:\n"
+                "\tbl Random\n"
+                "\tlsls r0, 16\n"
+                "\tlsrs r0, 16\n"
+                "\tmovs r1, 0xF\n"
+                "\tbl __umodsi3\n"
+                "\tadds r0, 0x1\n"
+                "_08146C7E:\n"
+                "\tstrb r0, [r5, 0x1]\n"
+                "_08146C80:\n"
+                "\tpop {r4,r5}\n"
+                "\tpop {r0}\n"
+                "\tbx r0");
+}
+#endif
+
+void sub_8146C88(void)
+{
+    struct MysteryEventStruct *r5 = sub_8143D94();
+
+    r5->unk_0_0 = 0;
+    r5->unk_0_2 = 0;
+    r5->unk_0_5 = 0;
+    r5->unk_1 = 0;
+    VarSet(0x4028, 0);
+}
+
+void sub_8146CA4(void)
+{
+    u16 *r4 = sub_806E454(0x4028);
+    struct MysteryEventStruct *r2 = sub_8143D94();
+    struct MysteryEventStruct r0 = *r2;
+
+    if ((u8)r0.unk_0_5 > 4 && ++(*r4) > 0x1f3)
+    {
+        r2->unk_0_5 = 0;
+        *r4 = 0;
+    }
+}
+
+u16 sub_8146CE8(void)
+{
+    u16 *r6 = &gUnknown_20370D0;
+    struct MysteryEventStruct *r4 = sub_8143D94();
+    u16 r5;
+
+    if (!sub_806E2BC() || !sub_8143E1C())
+        return 0;
+
+    r5 = sub_8146E0C(r4);
+
+    switch (r5)
+    {
+        case 0:
+            break;
+        case 1:
+            *r6 = sub_8146D74(r4);
+            break;
+        case 2:
+            *r6 = sub_8146D74(r4);
+            break;
+        case 3:
+            break;
+        case 4:
+            *r6 = sub_8146D74(r4);
+            sub_8146DA0(r4);
+            break;
+        case 5:
+            *r6 = sub_8146D74(r4);
+            sub_8146D94(r4);
+            break;
+        case 6:
+            break;
+    }
+
+    return r5;
+}
+
+static u32 sub_8146D74(struct MysteryEventStruct *a0)
+{
+    u32 r4;
+
+    a0->unk_0_0 = 0;
+    r4 = a0->unk_1 + 0x84;
+    a0->unk_1 = 0;
+    sub_8146DD8(a0);
+    return r4;
+}
+
+static void sub_8146D94(struct MysteryEventStruct *a0)
+{
+    a0->unk_0_2 = 0;
+}
+
+static void sub_8146DA0(struct MysteryEventStruct *a0)
+{
+    a0->unk_0_2++;
+    if ((u8)a0->unk_0_2 > 4)
+        a0->unk_0_2 = 4;
+}
+
+static void sub_8146DD8(struct MysteryEventStruct *a0)
+{
+    a0->unk_0_5++;
+    if ((u8)a0->unk_0_5 > 5)
```

Comment:
unnecessary cast?  It's already `u8`.  Same with `sub_8146E0C`

Reply:
It's necessary to match.